shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala

```diff
@@ -673,7 +674,8 @@ object GlutenConfig {
       // gcs config
       SPARK_GCS_STORAGE_ROOT_URL,
       SPARK_GCS_AUTH_TYPE,
-      SPARK_GCS_AUTH_SERVICE_ACCOUNT_JSON_KEYFILE
+      SPARK_GCS_AUTH_SERVICE_ACCOUNT_JSON_KEYFILE,
+      GLUTEN_REGEX_LOG_REDACTION
```

Comment:
@ArnavBalyan, I think we should just use the existing redaction config key in Spark. Thus, this Spark config will be respected in Gluten.

See the below example to pass Spark's session timezone:
https://github.com/apache/incubator-gluten/pull/6793/files#diff-21453c657b1ccfba45172821a434e5e7cdf007a49b440155514e6641b837d77cL651

Reply:
Thanks, I was trying to do this, then realized some compatibility problems with C++ `regex` library vs Scala `matching` library used by Spark. 
For example: Spark typically uses something like `(?i)secret\|password\|token\|access[.]key`, but this fails in C++ regex library because `(?i)` is not supported and users need to represent the regex in a different manner. To prevent such conflicts, for now I had created a different config specific to gluten. I agree, this is not the most elegant way, can switch to single config as well, let me know thanks!

Reply:
@ArnavBalyan, can we use boost::regex instead? Such pattern should be supported by boost.

Reply:
Yes we can use boost, there are still be parity differences, but these would be less common. Also just noticed the regex redact property is present in spark `core` config, it seems currently the `SQL` configs flow in, we might have to wire up the core configs to flow into gluten.

Reply:
@ArnavBalyan, yes, we cannot guarantee the compatibility for all regex pattern. But at least, this default regex should be supported in Gluten.

Reply:
yes theres differences in the c++ regex library, we could probably use a fallback to something sensible as a compromise.

Reply:
Thanks for the review @PHILO-HE @xushichangdesmond, I'll wire up the spark core config and use that by default, maybe we can improve this to fallback to gluten specific configs if needed in the future

Reply:
> For example: Spark typically uses something like (?i)secret\|password\|token\|access[.]key, but this fails in C++ regex library because (?i) is not supported and users need to represent the regex in a different manner.

could we overwrite Spark's default pattern with a valid regex patten in native?

Reply:
There are alot of incompatibilities, it's just one of the things where c++ and java don't have complete parity, I would rather, we provide limited support for the same config, or allow users to provide regex supported by c++ using a different properties. Option 1 is better for now and later we can add support